README.md
# switch 2 or 3 states
An easy way to implement a switch with 2 or 3 states.
In this package you also have the freedom to configure the list of values that will be used on the switch.<br/>

The list of values is dynamic and returns the selected value without having to do the conversion.
Example of the list of values:
```
[false,true]
[false,null,true]
[0,2,1]
['no','','yes']
```

## Getting Started
```
   WidgetSwitch1(
                iconRight: Icons.check,
                initValue: true,
                width: 200,
                buttonDiameter: 55,
                iconSize: 60,
                ///this is dymamic type list to values user parsed
                values: [false, null, true],
                ///values: [0,1,2],
                ///values: ['0','1','2'],
                ///values: [0,2,1],
                onChange: (x) {
                  print(x);                
                },
              ),
```

## Demo
![Demo](https://github.com/robertoltrocha/flutter_switch_2_3_states/blob/main/images/example1.gif)

## Contributions 
If you find a bug or want a feature, but don't know how to fix/implement it, please fill an [issue](https://github.com/robertoltrocha/flutter_switch_2_3_states/issues).

## Licence
MIT

CHANGELOG.md
## [0.0.9] 2021-06-06 
* Update dependecy cupertino_icons to 1.0.3

## [0.0.8] 2021-04-26 
| Note: This release has breaking changes.
* BREAKING: isThreeStates has been removed. Now, to set the switch with two states, parse the list with two values, example: values = [false, true]
* Add height option on WidgetSwitch1, see the effect in example 3.

## [0.0.7] 2021-04-23 
* Bug: WidgetSwitch3 - center text is disappearing when button move to left and right
* Changed default list [0, 1, 2] to [0, 2, 1]
* Added asserts into the classes
* Removed dart_style dependency

## [0.0.6] 2021-04-22 
* Dart format
* Update readme.md
* Add new WidgetSwitch3 

## [0.0.5] 2021-04-22 
* Update documents

## [0.0.4] 2021-04-22 
* Update documents

## [0.0.3] 2021-04-21 
| Note: This release has breaking changes.
* BREAKING: buttonDiameter has been renamed to buttonSize

## [0.0.2] 2021-04-21 
* Update readme.md

## [0.0.1] 2021-04-21 
* Initial.



test/flutter_switch_2_3_states_test.dart
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('adds one to input values', () {


  });
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_switch_2_3_states/flutter_switch_2_3_states.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Switch with 2 or 3 states',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Exemple switch with 2 or 3 states'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({required this.title});

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  dynamic value1 = 1;
  dynamic value2;
  dynamic value3 = 1;
  dynamic value4 = 2;
  dynamic value5 = 0;
  dynamic value6 = 'yes';
  dynamic value7 = 'yes';
  dynamic value8 = '';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(widget.title),
        ),
        body: Center(
          child: SingleChildScrollView(
            child: Container(
              constraints: BoxConstraints(
                  maxWidth: MediaQuery.of(context).size.width * 0.9),
              child: Column(
                children: [
                  SizedBox(height: 20),
                  Text(
                    'Ex.1 Switch 3 states default',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    onChange: (x) {
                      print(x);
                      setState(() {
                        value1 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value1.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.2 Switch 3 states with text and box button',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch3(
                    initValue: value8,
                    textLeft: "NO",
                    textCenter: "?",
                    textRight: "YES",
                    values: ['no', '', 'yes'],
                    onChange: (x) {
                      print(x);
                      setState(() {
                        value8 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value8.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.3 Switch 3 states button size greater than switch height',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    initValue: 0,
                    iconLeft: null,
                    iconRight: null,
                    height: 10,
                    buttonSize: 30,
                    onChange: (x) {
                      print(x);
                    },
                  ),
                  SizedBox(height: 20),
                  Text(
                    'Ex.4 Switch 3 states and change the size and color of the button',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    buttonSize: 55,
                    initValue: value2,
                    width: 200,
                    colorButton: Colors.blueGrey,

                    ///this is dymamic type list to values user parsed
                    values: [false, null, true],

                    ///values: [0,1,2],
                    ///values: ['0','1','2'],
                    ///values: [0,2,1],
                    onChange: (x) {
                      print(x);
                      setState(() {
                        value2 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value2.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.5 Switch 2 states',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    colorRightBackground: Colors.blue,
                    colorLeftBackground: Colors.orange,
                    iconLeft: Icons.remove_circle,
                    iconRight: Icons.play_arrow,
                    initValue: value3,
                    width: 70,
                    duration: 500,
                    values: [0, 1],
                    onChange: (x) {
                      setState(() {
                        value3 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value3.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.6 Switch 3 states and change duration time',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    colorRightBackground: Colors.blue,
                    colorLeftBackground: Colors.orange,
                    iconLeft: Icons.remove_circle,
                    iconRight: Icons.play_arrow,
                    initValue: value4,
                    duration: 1000,
                    values: [0, 2, 1],
                    onChange: (x) {
                      setState(() {
                        value4 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value4.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.7 Switch 3 states with text',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch2(
                    colorRightBackground: Colors.blue,
                    colorLeftBackground: Colors.orange,
                    initValue: value4,
                    values: [0, 2, 1],
                    onChange: (x) {
                      setState(() {
                        value4 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value4.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.8 Switch 2 states with text',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch2(
                    width: 70,
                    initValue: value5,
                    values: [0, 1],
                    onChange: (x) {
                      setState(() {
                        value5 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value5.toString()),
                  SizedBox(height: 20),
                  Text(
                    'Ex.9 Switch 3 states disabled',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch1(
                    enable: false,
                    initValue: null,
                    values: [false, null, true],
                    onChange: (x) {},
                  ),
                  SizedBox(height: 20),
                  Text(
                    'Ex.10 Switch 2 states with custom text',
                    style: TextStyle(fontWeight: FontWeight.bold),
                  ),
                  WidgetSwitch2(
                    width: 90,
                    initValue: value6,
                    colorRightBackground: Colors.orange,
                    textLeft: "NO",
                    textRight: "YES",
                    values: ['no', 'yes'],
                    onChange: (x) {
                      setState(() {
                        value6 = x;
                      });
                    },
                  ),
                  Text('the actual value is ' + value6.toString()),
                  SizedBox(height: 20),
                ],
              ),
            ),
          ),
        ));
  }
}


lib/widgets/widget_switch1.dart
import 'package:flutter/material.dart';

// ignore: must_be_immutable
class WidgetSwitch1 extends StatefulWidget {
  /// Key
  /// Key
  Key? key;

  /// return the value on change tap
  final ValueChanged<dynamic> onChange;

  /// background colorRight state
  late Color colorRightBackground;

  /// color rightIcon state
  late Color colorRightIcon;

  /// IconRight state
  final IconData? iconRight;

  /// colorLeftBackground state
  late Color colorLeftBackground;

  /// colorLeftIcon state
  late Color colorLeftIcon;

  /// icon left state
  final IconData? iconLeft;

  /// button color
  final Color colorButton;

  ///Set disable opacity
  late double disableOpacity;

  /// width switch
  final double width;

  /// height switch
  double? height;

  /// Background center state
  late Color backgroundCenter;

  /// Init value/default
  late dynamic initValue;

  /// List of values to 2 or 3 states
  final List<dynamic> values;

  /// set duration transition between states
  late int duration;

  /// set switch enable
  late bool enable;

  ///size button
  late double buttonSize;

  WidgetSwitch1(
      {this.key,
      this.colorLeftIcon = Colors.red,
      this.colorLeftBackground = Colors.red,
      this.iconLeft = Icons.close,
      this.colorRightIcon = Colors.green,
      this.colorRightBackground = Colors.green,
      this.iconRight = Icons.check,
      this.colorButton = const Color(0xffF0F0F0),
      this.width = 100,
      this.backgroundCenter = const Color(0xFFF0F0F0),
      required this.onChange,
      this.duration = 200,
      this.enable = true,
      this.disableOpacity = 0.6,
      this.buttonSize = 20,
      this.values = const [0, 2, 1],
      this.height,
      this.initValue = 2})
      : assert((values.length >= 2 && values.length <= 3),
            'You must provide a list with 2 or 3 values.'),
        assert(width >= 60),
        assert(disableOpacity >= 0 && disableOpacity <= 1),
        assert(duration >= 200 && duration <= 2000),
        super(key: key);

  @override
  _WidgetSwitch1State createState() => _WidgetSwitch1State();
}

class _WidgetSwitch1State extends State<WidgetSwitch1> {
  ///Color right icon
  late Color _colorRightIcon;

  ///Color left icon
  late Color _colorLeftIcon;

  ///the result actual select state value
  int _toggleIndex = 1;

  ///Set all state to false, not used active states
  late List<bool> _selections;

  ///padding left and right side of button
  double _rPos = 0;

  ///padding left and right side of button
  double _padding = 4;

  ///  Color ColorCenter = Colors.grey;
  Color _background = Colors.grey;

  /// widget switch height
  double _height = 0;

  @override
  void initState() {
    super.initState();
    widget.height == null
        ? _height = widget.buttonSize + 8
        : _height = widget.height!;
    _selections = List.generate(widget.values.length, (_) => false);
    _colorRightIcon = widget.colorRightIcon;
    _colorLeftIcon = widget.colorLeftIcon;
    _toggleIndex = widget.values.indexOf(widget.initValue) == -1
        ? 0
        : widget.values.indexOf(widget.initValue);
    _move(_toggleIndex);
  }

  @override
  Widget build(BuildContext context) {
    return widget.values.length == 3 ? _switch3Pos() : _switch2Pos();
  }

  ///build switch with 3 states
  Widget _switch3Pos() {
    return Opacity(
      opacity: widget.enable ? 1 : widget.disableOpacity,
      child: Stack(alignment: Alignment.center, children: [
        Container(
          width: widget.width,
          height:
              widget.buttonSize > _height ? widget.buttonSize : widget.height,
        ),
        Container(
          width: widget.width,
          decoration: BoxDecoration(
              color: _background, borderRadius: BorderRadius.circular(40)),
          constraints: BoxConstraints(maxHeight: _height),
        ),
        AnimatedContainer(
          padding: EdgeInsets.zero,

          duration: Duration(milliseconds: widget.duration),

          /// Provide an optional curve to make the animation feel smoother.
          curve: Curves.fastOutSlowIn,
          width: widget.width,
          height: widget.buttonSize,

          decoration: BoxDecoration(
              color: Colors.transparent,
              borderRadius: BorderRadius.circular(40)),
          child: ToggleButtons(
              borderColor: Colors.transparent,
              selectedColor: Colors.transparent,
              renderBorder: false,
              constraints: BoxConstraints.expand(width: widget.width / 3),

              ///number 3 is number of toggle buttons
              children: [
                Icon(
                  widget.iconLeft,
                  color: _colorLeftIcon,
                  size: widget.buttonSize,
                ),
                Container(
                  padding: EdgeInsets.zero,
                ),
                Icon(
                  widget.iconRight,
                  color: _colorRightIcon,
                  size: widget.buttonSize,
                ),
              ],
              onPressed: widget.enable
                  ? (int index) {
                      _changeSelect(index);
                    }
                  : null,
              borderRadius: BorderRadius.circular(20),
              borderWidth: 3,
              isSelected: _selections),
        ),
        AnimatedPositioned(
            top: widget.buttonSize < _height
                ? (_height - widget.buttonSize) / 2
                : 0,
            left: _rPos,
            child: _iconButton(),
            duration: Duration(milliseconds: widget.duration)),
      ]),
    );
  }

  ///build switch with 2 states
  Widget _switch2Pos() {
    return Opacity(
      opacity: widget.enable ? 1 : widget.disableOpacity,
      child: Stack(children: [
        AnimatedContainer(
          duration: Duration(milliseconds: 500),

          /// Provide an optional curve to make the animation feel smoother.
          curve: Curves.fastOutSlowIn,
          constraints: BoxConstraints(minHeight: widget.buttonSize + 8),
          width: widget.width,
          decoration: BoxDecoration(
              color: _background, borderRadius: BorderRadius.circular(20)),
          child: LayoutBuilder(builder: (context, constraints) {
            return ToggleButtons(
                borderColor: Colors.transparent,
                selectedColor: Colors.transparent,
                renderBorder: false,
                constraints:
                    BoxConstraints.expand(width: constraints.maxWidth / 2),

                ///number 2 is number of toggle buttons
                children: [
                  Icon(
                    widget.iconLeft,
                    color: _colorLeftIcon,
                    size: widget.buttonSize,
                  ),
                  Icon(
                    widget.iconRight,
                    color: _colorRightIcon,
                    size: widget.buttonSize,
                  ),
                ],
                onPressed: widget.enable
                    ? (int index) {
                        _changeSelect(index);
                      }
                    : null,
                borderRadius: BorderRadius.circular(20),
                borderWidth: 3,
                isSelected: _selections);
          }),
        ),
        AnimatedPositioned(
            top: 4,
            left: _rPos,
            child: _iconButton(),
            duration: Duration(milliseconds: widget.duration)),
      ]),
    );
  }

  ///build button
  Widget _iconButton() {
    return Material(
        elevation: 2,
        type: MaterialType.button,
        color: Colors.transparent,
        borderRadius: BorderRadius.circular(1000),
        child: Container(
          height: widget.buttonSize,
          width: widget.buttonSize,
          decoration: BoxDecoration(
            color: widget.colorButton,
            border: Border.all(color: Color(0xF0F0F0), width: 2.0),
            borderRadius: BorderRadius.circular(1000),
          ),
        ));
  }

  ///move button
  void _move(int index) {
    _toggleIndex = index;
    setState(() {
      if (widget.values.length == 3) {
        switch (_toggleIndex) {
          case 0:
            _rPos = _padding;
            _background = widget.colorLeftBackground;
            _colorRightIcon = Colors.transparent;
            _colorLeftIcon = Colors.transparent;
            break;
          case 1:
            _rPos = widget.width / 2 - widget.buttonSize / 2;
            _background = widget.backgroundCenter;
            _colorRightIcon = widget.colorRightIcon;
            _colorLeftIcon = widget.colorLeftIcon;
            break;
          case 2:
            _rPos = widget.width - widget.buttonSize - _padding;
            _background = widget.colorRightBackground;
            _colorLeftIcon = Colors.transparent;
            _colorRightIcon = Colors.transparent;
            break;
          default:
        }
      } else {
        switch (_toggleIndex) {
          case 0:
            _rPos = _padding;
            _background = widget.colorLeftBackground;
            _colorRightIcon = Colors.transparent;
            _colorLeftIcon = Colors.transparent;
            break;
          case 1:
            _rPos = widget.width - widget.buttonSize - _padding;
            _background = widget.colorRightBackground;
            _colorLeftIcon = Colors.transparent;
            _colorRightIcon = Colors.transparent;
            break;
          default:
        }
      }
    });
  }

  ///return value after selected change
  void _changeSelect(int index) {
    _move(index);
    widget.onChange(widget.values[_toggleIndex]);
  }
}


lib/widgets/widget_switch3.dart
import 'package:flutter/material.dart';

// ignore: must_be_immutable
class WidgetSwitch3 extends StatefulWidget {
  ///Key
  Key? key;

  /// return the value on change tap
  final ValueChanged<dynamic> onChange;

  ///Right text
  late String textRight;

  ///Left Text
  late String textLeft;

  ///Left Text
  late String textCenter;

  /// background colorRight state
  late Color colorRightBackground;

  /// colorLeftBackground state
  late Color colorLeftBackground;

  ///Set disable opacity
  late double disableOpacity;

  /// color button
  final Color colorButton;

  /// width switch
  final double width;

  /// Background center state
  late Color backgroundCenter;

  /// Init value/default
  late dynamic initValue;

  ///List of values to 2 or 3 states
  final List<dynamic> values;

  ///set duration transition between states
  late int duration;

  ///switch enable
  late bool enable;

  ///button size
  late double buttonSize;

  WidgetSwitch3(
      {this.key,
      this.colorLeftBackground = Colors.red,
      this.disableOpacity = 0.6,
      this.colorRightBackground = Colors.green,
      this.colorButton = const Color(0xffF0F0F0),
      this.width = 120,
      this.backgroundCenter = const Color(0xFFF0F0F0),
      required this.onChange,
      this.duration = 200,
      this.buttonSize = 40,
      this.enable = true,
      this.textLeft = "Off",
      this.textRight = "On",
      this.textCenter = "?",
      this.values = const [0, 2, 1],
      this.initValue = 2})
      : assert((values.length >= 2 && values.length <= 3),
            'You must provide a list with 2 or 3 values.'),
        assert(width >= 60),
        assert(disableOpacity >= 0 && disableOpacity <= 1),
        assert(duration >= 200 && duration <= 2000),
        super(key: key);

  @override
  _WidgetSwitch3State createState() => _WidgetSwitch3State();
}

class _WidgetSwitch3State extends State<WidgetSwitch3> {
  late Color colorRightBackground;
  late Color colorLeftBackground;
  late Color backgroundDefault;
  String textRight = "";
  String textLeft = "";
  String textButton = "";

  ///the result actual select state value
  int toggleIndex = 1;

  ///Set all state to false, not used active states
  late List<bool> _selections;

  ///padding left and right side of button
  double rPos = 0;

  ///padding left and right side of button
  double padding = 4;

  ///  Color ColorCenter = Colors.grey;
  Color background = Colors.grey;

  ///Text color / left, right, center e button text
  Color textRightColor = Colors.black;
  Color textLeftColor = Colors.black;
  Color textCenterColor = Colors.black;
  Color textButtonColor = Colors.black;

  ///Text button color active
  Color textButtonColorActive = Colors.white;

  @override
  void initState() {
    super.initState();
    _selections = List.generate(widget.values.length, (_) => false);

    textLeft = widget.textLeft;
    textRight = widget.textRight;

    toggleIndex = widget.values.indexOf(widget.initValue) == -1
        ? 0
        : widget.values.indexOf(widget.initValue);
    _move(toggleIndex);
  }

  @override
  Widget build(BuildContext context) {
    return widget.values.length == 3 ? _switch3Pos() : _switch2Pos();
  }

  Widget _switch3Pos() {
    return Opacity(
      opacity: widget.enable ? 1 : widget.disableOpacity,
      child: Stack(children: [
        AnimatedContainer(
          constraints: BoxConstraints(minHeight: widget.buttonSize + 8),
          duration: Duration(milliseconds: widget.duration),

          /// Provide an optional curve to make the animation feel smoother.
          curve: Curves.fastOutSlowIn,
          width: widget.width,
          decoration: BoxDecoration(
            color: Color(0XFFF0F0F0),
          ),
          child: LayoutBuilder(builder: (context, constraints) {
            return ToggleButtons(
                selectedColor: Colors.transparent,
                color: Colors.transparent,
                renderBorder: false,

                ///number 3 is number of toggle buttons
                constraints:
                    BoxConstraints.expand(width: constraints.maxWidth / 3),
                children: [
                  Text(
                    textLeft,
                    style: TextStyle(
                      color: textLeftColor,
                    ),
                    textAlign: TextAlign.center,
                  ),
                  Container(
                    child: Text(
                      widget.textCenter,
                      style: TextStyle(color: textCenterColor),
                      textAlign: TextAlign.center,
                    ),
                    padding: EdgeInsets.zero,
                  ),
                  Container(
                    child: Text(
                      textRight,
                      style: TextStyle(color: textRightColor),
                      textAlign: TextAlign.center,
                    ),
                    padding: EdgeInsets.zero,
                  ),
                ],
                onPressed: widget.enable
                    ? (int index) {
                        _changeSelect(index);
                      }
                    : null,
                borderRadius: BorderRadius.circular(20),
                borderWidth: 3,
                isSelected: _selections);
          }),
        ),
        AnimatedPositioned(
            top: 4,
            left: rPos,
            child: _iconButton(),
            duration: Duration(milliseconds: widget.duration)),
      ]),
    );
  }

  Widget _switch2Pos() {
    return Opacity(
      opacity: widget.enable ? 1 : widget.disableOpacity,
      child: Stack(children: [
        AnimatedContainer(
          constraints: BoxConstraints(minHeight: widget.buttonSize + 8),
          duration: Duration(milliseconds: widget.duration),

          /// Provide an optional curve to make the animation feel smoother.
          curve: Curves.fastOutSlowIn,
          width: widget.width,
          decoration: BoxDecoration(
            color: Color(0XFFF0F0F0),
          ),
          child: LayoutBuilder(builder: (context, constraints) {
            return ToggleButtons(
                selectedColor: Colors.transparent,
                color: Colors.transparent,
                renderBorder: false,

                ///number 3 is number of toggle buttons
                constraints:
                    BoxConstraints.expand(width: constraints.maxWidth / 2),
                children: [
                  Text(
                    textLeft,
                    style: TextStyle(
                      color: textLeftColor,
                    ),
                    textAlign: TextAlign.center,
                  ),
                  Text(
                    textRight,
                    style: TextStyle(color: textRightColor),
                    textAlign: TextAlign.center,
                  ),
                ],
                onPressed: widget.enable
                    ? (int index) {
                        _changeSelect(index);
                      }
                    : null,
                borderRadius: BorderRadius.circular(20),
                borderWidth: 3,
                isSelected: _selections);
          }),
        ),
        AnimatedPositioned(
            top: 4,
            left: rPos,
            child: _iconButton(),
            duration: Duration(milliseconds: widget.duration)),
      ]),
    );
  }

  Widget _iconButton() {
    double bevel = 4;
    Offset blurOffset = new Offset(2, 2);

    return Container(
      height: widget.buttonSize,
      width: widget.buttonSize,
      decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(bevel),
          color: background,
          boxShadow: [
            BoxShadow(
              blurRadius: bevel,
              offset: -blurOffset,
              color: background,
            ),
            BoxShadow(
                blurRadius: bevel,
                offset: blurOffset,
                color: Colors.grey.shade400)
          ]

          // borderRadius: BorderRadius.circular(12.0),
          ),
      child: Center(
        child: Text(
          textButton,
          style: TextStyle(color: textButtonColorActive),
        ),
      ),
    );
  }

  void _move(int index) {
    toggleIndex = index;
    setState(() {
      if (widget.values.length == 3) {
        switch (toggleIndex) {
          case 0:
            rPos = padding;
            background = widget.colorLeftBackground;
            textRightColor = Colors.black;
            textLeftColor = Colors.white;
            textButton = widget.textLeft;

            textButtonColorActive = Colors.white;
            break;
          case 1:
            rPos = widget.width / 2 - widget.buttonSize / 2;
            background = widget.backgroundCenter;
            textRight = widget.textRight;
            textLeft = widget.textLeft;
            textRightColor = Colors.black;
            textLeftColor = Colors.black;
            textButtonColorActive = Colors.black;
            textButton = widget.textCenter;
            break;
          case 2:
            rPos = widget.width - widget.buttonSize - padding;
            background = widget.colorRightBackground;

            textRightColor = Colors.white;
            textLeftColor = Colors.black;
            textButton = widget.textRight;
            textButtonColorActive = Colors.white;

            break;
          default:
        }
      } else {
        switch (toggleIndex) {
          case 0:
            rPos = padding;
            background = widget.colorLeftBackground;
            textRightColor = Colors.black;
            textLeftColor = Colors.white;
            textButton = widget.textLeft;
            break;
          case 1:
            rPos = widget.width - widget.buttonSize - padding;
            background = widget.colorRightBackground;
            textRightColor = Colors.white;
            textLeftColor = Colors.black;
            textButton = widget.textRight;
            break;
          default:
        }
      }
    });
  }

  void _changeSelect(int index) {
    _move(index);
    widget.onChange(widget.values[toggleIndex]);
  }
}
